search_test.go

```diff
@@ -29,3 +29,21 @@ func TestNewEntry(t *testing.T) {
 		iteration = iteration + 1
 	}
 }
+
+func TestGetAttributeValue(t *testing.T) {
+	dn := "testDN"
+	attributes := map[string][]string{
+		"Alpha":   {"value"},
```

Comment:
what should happen if you have `{"alpha":{"a","b"}, "ALPHA": {"c", "d"}}` and you search for "Alpha"?

Reply:
is that possible? isn't the attribute name case insensitive?

Reply:
Shall we at least add a test case with attribute names differing only in case and document via the tests how we'd handle it? That way at least it's more obvious.